single-pool/program/tests/initialize.rs

```diff
@@ -56,4 +61,56 @@ async fn fail_double_init() {
     check_error(e, SinglePoolError::PoolAlreadyInitialized);
 }
 
+#[test_case(true; "minimum_enabled")]
+#[test_case(false; "minimum_disabled")]
+#[tokio::test]
+async fn fail_below_pool_minimum(enable_minimum_delegation: bool) {
+    let mut context = program_test(enable_minimum_delegation)
+        .start_with_context()
+        .await;
+    let accounts = SinglePoolAccounts::default();
+    let slot = context.genesis_config().epoch_schedule.first_normal_slot + 1;
+    context.warp_to_slot(slot).unwrap();
+
+    create_vote(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+        &accounts.validator,
+        &accounts.voter.pubkey(),
+        &accounts.withdrawer.pubkey(),
+        &accounts.vote_account,
+    )
+    .await;
+
+    let rent = context.banks_client.get_rent().await.unwrap();
+    let minimum_delegation = get_pool_minimum_delegation(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+    )
+    .await;
+
+    let instructions = instruction::initialize(
+        &id(),
+        &accounts.vote_account.pubkey(),
+        &context.payer.pubkey(),
+        &rent,
+        minimum_delegation - 1,
+    );
```

Comment:
Maybe this is already tested elsewhere, but rather than checking against `minimum_delegation - 1`, would it be more useful to test `LAMPORTS_PER_SOL - 1` and make sure that it fails in *both* cases? Then you're sure that the pool minimum is respected

Reply:
oh `get_pool_minimum_delegation` is a helper that returns `max(stake_program_minimum_delegation, LAMPORTS_PER_SOL)`, using the constant would be clearer tho

Reply:
Oh right, oops, all good then!